TaskWithInheritanceAndPolymorphism/TaskWithInheritanceAndPolymorphism.cpp

```diff
@@ -0,0 +1,135 @@
+﻿#include <iostream>
+#include <memory>
+#include <vector>
+
+template <typename T> class Figure
+{
+public:
+	Figure()
+	{
+		std::cout << "Figure()" << std::endl;
+	}
+	~Figure()
+	{
+		std::cout << "~Figure()" << std::endl;
+	}
+
+	virtual void what() const = 0;
+	virtual double area() const = 0;
+	virtual double perimeter() const = 0;
+};
+
+class Square final : public Figure<double>
```

Comment:
The Square and others should be a template as well as the Figure

Reply:
Used